.github/scripts/fetch_latest_green_commit.py

```diff
@@ -0,0 +1,156 @@
+            "log",
+            "-n",
+            "1",
+            "--pretty=format:%H",
+            "origin/viable/strict",
+        ],
+        encoding="ascii",
+    )
+    commits = _check_output(
+        [
+            "git",
+            "rev-list",
+            f"{latest_viable_commit}^..HEAD",
+            "--remotes=*origin/main",
+        ],
+        encoding="ascii",
+    ).splitlines()
+
+    return commits
+
+
+def query_commits(commits: List[str]) -> List[Dict[str, Any]]:
+    rs = rockset.RocksetClient(
+        host="api.usw2a1.rockset.com", api_key=os.environ["ROCKSET_API_KEY"]
+    )
+    params = [{"name": "shas", "type": "string", "value": ",".join(commits)}]
+    res = rs.QueryLambdas.execute_query_lambda(
+        # https://console.rockset.com/lambdas/details/commons.commit_jobs_batch_query
+        query_lambda="commit_jobs_batch_query",
+        version="19c74e10819104f9",
+        workspace="commons",
+        parameters=params,
+    )
+
+    return cast(List[Dict[str, Any]], res.results)
+
+
+def print_commit_status(commit: str, results: Dict[str, Any]) -> None:
+    print(commit)
+    for check in results["results"]:
+        if check["sha"] == commit:
+            print(f"\t{check['conclusion']:>10}: {check['name']}")
+
+
+def get_commit_results(
+    commit: str, results: List[Dict[str, Any]]
+) -> List[Dict[str, Any]]:
+    workflow_checks = []
+    for check in results:
+        if check["sha"] == commit:
+            workflow_checks.append(
+                WorkflowCheck(
+                    workflowName=check["workflowName"],
+                    name=check["name"],
+                    jobName=check["jobName"],
+                    conclusion=check["conclusion"],
+                )._asdict()
+            )
+    return workflow_checks
+
+
+def is_green(
+    commit: str, requires: List[str], results: List[Dict[str, Any]]
+) -> Tuple[bool, str]:
+    workflow_checks = get_commit_results(commit, results)
+
+    regex = {name: False for name in requires}
+
+    for check in workflow_checks:
+        jobName = check["jobName"]
+        # Ignore result from unstable job, be it success or failure
+        if "unstable" in jobName:
+            continue
+
+        workflowName = check["workflowName"]
+        conclusion = check["conclusion"]
+        for required_check in regex:
+            if re.match(required_check, workflowName, flags=re.IGNORECASE):
+                if conclusion not in ["success", "skipped"]:
+                    return (False, workflowName + " checks were not successful")
+                else:
+                    regex[required_check] = True
+
+    missing_workflows = [x for x in regex.keys() if not regex[x]]
+    if len(missing_workflows) > 0:
+        return (False, "missing required workflows: " + ", ".join(missing_workflows))
+
+    return (True, "")
+
+
+def get_latest_green_commit(
+    commits: List[str], requires: List[str], results: List[Dict[str, Any]]
+) -> Any:
```

Comment:
nit: Would be Optional[str] better since the return type is either None or a str